libs/methodologies/bold/rule-processors/credits/src/rewards-distribution/rewards-distribution.test-cases.ts

```diff
@@ -0,0 +1,637 @@
+      amount: '22.9995',
+      percentage: '15',
+    },
+    {
+      actorType: RewardsDistributionActorType.NETWORK,
+      amount: '30.666',
+      percentage: '20',
+    },
+    {
+      actorType: RewardsDistributionActorType.METHODOLOGY_AUTHOR,
+      amount: '1.5333',
+      percentage: '1',
+    },
+    {
+      actorType: RewardsDistributionActorType.METHODOLOGY_DEVELOPER,
+      amount: '1.5333',
+      percentage: '1',
+    },
+    {
+      actorType: MethodologyActorType.REMAINDER,
+      amount: '0',
+      percentage: '0',
+    },
+  ],
+};
+
+export const rewardsDistributionProcessorTestCases: TestCase[] = [
+  {
+    certificateDocuments: documents.standard.certificateDocuments,
+    creditsDocument: documents.standard.creditsDocument,
+    expectedActorsResult: expectedResults.singleCertificateStandard,
+    expectedCertificateTotalValue: documents.massId1Value,
+    resultStatus: RuleOutputStatus.APPROVED,
+    scenario: 'single certificate with equal distribution',
+    unitPrice: UNIT_PRICE_VALUE,
+  },
+  {
+    certificateDocuments: documents.multiHauler.certificateDocuments,
+    creditsDocument: documents.multiHauler.creditsDocument,
+    expectedActorsResult: expectedResults.multipleHaulers,
+    expectedCertificateTotalValue: documents.massId1Value,
+    resultStatus: RuleOutputStatus.APPROVED,
+    scenario: 'single certificate with multiple HAULER participants',
+    unitPrice: UNIT_PRICE_VALUE,
+  },
+  {
+    certificateDocuments: documents.multipleCertificates.certificateDocuments,
+    creditsDocument: documents.multipleCertificates.creditsDocument,
+    expectedActorsResult: expectedResults.multipleCertificates,
+    expectedCertificateTotalValue: documents.multipleCertificates.totalValue,
+    resultStatus: RuleOutputStatus.APPROVED,
+    scenario: 'multiple certificates with the same participants',
+    unitPrice: UNIT_PRICE_VALUE,
+  },
+];
+
+const createErrorTestCases = () => {
+  const errorStubs = new BoldStubsBuilder()
+    .createMassIdDocuments()
+    .createMassIdAuditDocuments()
+    .createCertificateDocuments({
+      partialDocument: {
+        type: RECYCLED_ID,
+      },
+    })
+    .createCreditsDocument()
+    .build();
+
+  const creditsDocumentWithoutRulesMetadata = {
+    ...errorStubs.creditsDocument,
+    externalEvents: errorStubs.creditsDocument.externalEvents?.filter(
+      (event) => event.name !== RULES_METADATA.toString(),
+    ),
+  };
+
+  const certificateDocumentWithoutRulesMetadata = {
+    ...errorStubs.certificateDocuments[0]!,
+    externalEvents: errorStubs.certificateDocuments[0]!.externalEvents?.filter(
+      (event) => event.name !== RULES_METADATA.toString(),
+    ),
+  };
+
+  return {
+    certificateDocumentWithoutRulesMetadata,
+    creditsDocumentWithoutRulesMetadata,
+    errorStubs,
+  };
+};
+
+const errorTestData = createErrorTestCases();
+
+export const rewardsDistributionProcessorErrors: ErrorTestCase[] = [
+  {
+    certificateDocuments: [],
```

Comment:
To avoid ambiguity, I would rename usages of `certificate` to `massIdCertificate`
```suggestion
    massIdCertificateDocuments: [],
```

Reply:
https://github.com/carrot-foundation/methodology-rules/pull/222/commits/7a64e08980ee6a4d0c6d63d8b5a9e001e5d99f96